src/configuration.cpp

```diff
@@ -44,6 +45,23 @@ bool parse_url(const std::string &url, std::vector<std::optional<std::string>> &
 	return true;
 }
 
+std::string url_decode(const std::string &str) {
+	static const std::regex r(R"(%[0-9A-Fa-f]{2})", std::regex::extended);
+
+	std::stringstream ss;
+	std::smatch m;
+	for (int i = 0; i < str.length(); ++i) {
+		std::string substr = str.substr(i, 3);
+		if (std::regex_match(substr, m, r)) {
```

Comment:
A regex seems overkill here and this approach will silently ignore invalid percent-encoded characters, but it works for valid inputs.